### ISSUE TITLE
Use namespace packages

### ISSUE THREAD

**Issue by stasm** [projectfluent/python-fluent#79]
In preparation for #67 (or as a follow-up), I'd like to change the directory structure of this repository and start distributing Fluent code as [namespaced packages](https://packaging.python.org/guides/packaging-namespace-packages):

> Namespace packages allow you to split the sub-packages and modules within a single package across multiple, separate distribution packages

Here's the directory structure I suggested in https://github.com/projectfluent/python-fluent/pull/67#issuecomment-435599510:

```
fluent-syntax
    setup.py
    fluent
        __init__.py
        syntax
            __init__.py
            ...
fluent-bundle
    setup.py
    fluent
        __init__.py
        bundle
            __init__.py
            ...
```

**Comment by spookylukey** [comment]
As I said before, I don't have any experience with namespace packages, but I'm happy to go with this proposal. If separate repos are a better option, we can always do that later.

I'd suggest doing this change first, before merging #67. Then, I can adapt #67 to depend on a specific, older version of fluent. This will be a test of our ability to manage two packages within one repo.

Where do tests fit into this? It seems like there should be two separate test directories. We could have tox/travis set to run them as different 'environments', so that failures in one package are distinct from those in the other.

**Comment by spookylukey** [comment]
Based on my efforts in #80 and #81 - this seems to work. I've got my `FluentBundle` implementation that now depends on a specific, older version of `fluent` (which will soon be renamed to `fluent-syntax`), and tests run against that. Travis runs both sets of test suites for all environments, but separated out so that you can see which ones are failing more easily - https://travis-ci.org/projectfluent/python-fluent/builds/478585904

**Comment by stasm** [comment]
Nice. I've checked out your branch for #81 and tested locally with `fluent.syntax`, `fluent.bundle` and with the unpublished `fluent.migrate` module from https://hg.mozilla.org/l10n/fluent-migration. All worked great (with some additional dependencies installed for `fluent.bundle`; I'll test again when I review #81).

**Comment by stasm** [comment]
This was fixed by @spookylukey in #80. Since then, we published `fluent.syntax` 0.10 and `fluent.runtime` 0.1.
